src/uct/ib/mlx5/rc/rc_mlx5_common.c

```diff
@@ -620,34 +624,23 @@ uct_rc_mlx5_dp_ordering_ooo_init(uct_rc_mlx5_iface_common_t *iface,
      * - if dp_ordering_ooo is set, QPC/DCTC can enable AR.
      * - if dp_ordering_ooo_force is set, QPC/DCTC can request mlxreg
      *   configuration override, useful to force disable.
-     *
-     * QP modify behavior with returned values:
-     * - UCS_AUTO: Do not affect existing system behavior.
-     * - UCS_NO  : Force AR disabling on the QP if supported. QP modify will
-     *   return error on failure.
-     * - UCS_TRY : Set AR to enable, ignored if any failure.
-     * - UCS_YES : Force AR enabling on the QP if supported. QP modify will
-     *   return error on failure.
      */
 
-    if ((config->super.ar_enable == UCS_TRY) && dp_ordering_ooo) {
-        iface->super.super.config.dp_ordering_ooo = UCS_TRY;
-    } else if (config->super.ar_enable == UCS_NO) {
-        if (!dp_ordering_ooo_force) {
-            goto failure;
-        }
+    iface->config.force_ordering = ucs_ternary_auto_value_is_yes_or_no(
+            config->super.ar_enable);
 
-        iface->super.super.config.dp_ordering_ooo = UCS_NO;
-    } else if (config->super.ar_enable == UCS_YES) {
-        if (!dp_ordering_ooo_force || !dp_ordering_ooo) {
-            goto failure;
-        }
+    if ((/* Want to force configuration but can't force it*/
+         iface->config.force_ordering && !dp_ordering_ooo_force) ||
+        /* Want to force enable of adaptive routing but its unavailable */
+        ((config->super.ar_enable == UCS_YES) && !dp_ordering_ooo)) {
+        goto failure;
+    }
 
-        iface->super.super.config.dp_ordering_ooo = UCS_YES;
-    } else {
-        iface->super.super.config.dp_ordering_ooo = UCS_AUTO;
+    if (dp_ordering_ooo) {
```

Comment:
i understand that `dp_ordering_ooo` is the value reported in cap by FW.

in case ar_enable=no or auto, we need to set `ordering_level` to 0 (IBTA) even if md supports ar, otherwise we cannot force disable anymore and/or ar is always requested/enabled.